### PR TITLE
Upgrade ldns to 1.8.3 for Mariner 3.0

### PR DESCRIPTION
**Pull request by jslobodzian**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x]  `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Upgrades the ldns package to 1.8.3 for Mariner 3.0.  Updated the license name to SPDX standard.  Removed unnecessary or unsupported patches.

###### Does this affect the toolchain?  
**NO**

###### Test Methodology
There is no self test supported for this package.  Performed minimal validation by installing main package and utils package.  Spot checked a couple of binaries by running them and verifying that either outputs were generated or that a help page was provided.

### CODE REVIEW

File: SPECS/ldns/ldns.spec [hunk 1]
```diff
@@ -295,6 +280,12 @@ rm -rf doc/man
 %doc doc
 
 %changelog
+* Thu Jan 25 2024 CBL-Mariner Servicing Account <cblmargh@microsoft.com> - 1.8.3-1
+- Auto-upgrade to 1.8.3 - Upgrade for Azure Linux 3.0\
+- Removed unsupported multilib patch (azl only supports 64bit)
+- Removed other patches already integrated into this version
+- Updated license name to SPDX standard.
```
Comment (PawelWMS) [inline SPECS/ldns/ldns.spec:287]: Nice!

File: SPECS/ldns/ldns.spec [hunk 2]
```diff
@@ -295,6 +280,12 @@ rm -rf doc/man
 %doc doc
 
 %changelog
+* Thu Jan 25 2024 CBL-Mariner Servicing Account <cblmargh@microsoft.com> - 1.8.3-1
+- Auto-upgrade to 1.8.3 - Upgrade for Azure Linux 3.0\
```
Comment (PawelWMS) [inline SPECS/ldns/ldns.spec:284]: Nit-pick:

```suggestion
- Auto-upgrade to 1.8.3 - Upgrade for Azure Linux 3.0
```

File: SPECS/ldns/ldns-swig-4.2.patch [hunk 3]
```diff
@@ -0,0 +1,60 @@
+SWIG_Python_str_AsChar removal in SWIG 4.2.0
+
+The replacement, SWIG_PyUnicode_AsUTF8AndSize, has different memory
+management requirements.
+
+Submitted upstream: <https://github.com/NLnetLabs/ldns/pull/232>
+
+diff -ur ldns-1.8.3.orig/ldns-1.8.3/contrib/python/ldns_rdf.i ldns-1.8.3/ldns-1.8.3/contrib/python/ldns_rdf.i
+--- ldns-1.8.3.orig/ldns-1.8.3/contrib/python/ldns_rdf.i	2022-08-15 12:38:22.000000000 +0200
++++ ldns-1.8.3/ldns-1.8.3/contrib/python/ldns_rdf.i	2024-01-26 11:28:52.983540469 +0100
```
Comment (PawelWMS) [inline SPECS/ldns/ldns-swig-4.2.patch:10]: Can we do a few things:
- Apply the suggestion below to replace the `-p2` with `-p1` in the spec?
- Remove `%autopatch -p2` from the spec and add `-p1` into `%autosetup`?
- Remove `-N` from `%autosetup`?

Most patches come in a form, which makes them applicable with `-p1`, so it's probably going to make maintenance easier and keep the spec more consistent with most of our other ones.
```suggestion
--- ldns-1.8.3/contrib/python/ldns_rdf.i	2022-08-15 12:38:22.000000000 +0200
+++ ldns-1.8.3/contrib/python/ldns_rdf.i	2024-01-26 11:28:52.983540469 +0100
```

File: SPECS/ldns/ldns.spec [hunk 4]
```diff
@@ -32,48 +32,39 @@
...
 Name:           ldns
-Version:        1.7.0
-Release:        32%{?dist}
-License:        BSD
+Version:        1.8.3
+Release:        1%{?dist}
+License:        BSD-3-Clause
 Vendor:         Microsoft Corporation
-Distribution:   Mariner
+Distribution:   Azure Linux
```
Comment (PawelWMS) [inline SPECS/ldns/ldns.spec:39]: I completely forgot about the `Distribution` tag. Do we have a separate task to update these in bulk?

File: SPECS/ldns/ldns.spec [hunk 5]
```diff
@@ -32,48 +32,39 @@
...
 %if %{with perl}
 BuildRequires: perl-devel
 BuildRequires: perl-ExtUtils-MakeMaker
 BuildRequires: perl-generators
 BuildRequires: perl(Devel::CheckLib)
 %endif
+%if %{with python3}
+BuildRequires: python3-devel
+BuildRequires: swig
+%endif
```
Comment (PawelWMS) [inline SPECS/ldns/ldns.spec:62]: I think we may as well drop all macros for `python3` and just assume they are true. I don't think we're ever going to be building without it (until it gets deprecated and we'll have a massive migration anyway).;)
```suggestion
BuildRequires: python3-devel
BuildRequires: swig
```

File: SPECS/ldns/ldns.spec [hunk 6]
```diff
@@ -32,48 +32,39 @@
...
 Requires: ca-certificates
+Requires: glibc
+Requires: openssl
+
+Patch1: ldns-swig-4.2.patch
```
Comment (PawelWMS) [inline SPECS/ldns/ldns.spec:67]: Can you move the `Patch` tags below the `Source` ones? That's where we usually keep and expect them.

File: SPECS/ldns/ldns.spec [hunk 7]
```diff
@@ -32,48 +32,39 @@
...
 Group:          System Environment/Libraries
 Url:            http://www.nlnetlabs.nl/%{name}/
 Source0:        http://www.nlnetlabs.nl/downloads/%{name}/%{name}-%{version}.tar.gz
-Patch1:         ldns-1.7.0-multilib.patch
-Patch2:         ldns-1.7.0-parse-limit.patch
-Patch3:         ldns-1.7.0-realloc.patch
-Patch4:         ldns-1.7.0-Update-for-SWIG-4.patch
 
-# Only needed for builds from svn snapshot
-%if 0%{snapshot}
```
Comment (PawelWMS) [inline SPECS/ldns/ldns.spec:49]: If we're removing all uses of `%{snapshot}`, I think we should also remove this line at the top of the spec:
```
%{?!snapshot:         %global snapshot        0}
```

**Review by PawelWMS — COMMENTED** [review]
@jslobodzian, could you also update the title of the PR?

File: SPECS/ldns/ldns.spec (hunk 6 above)
Comment (jslobodzian) [reply SPECS/ldns/ldns.spec:67]: done

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**